tests/installation/upgrade_select_opensuse.pm

```diff
@@ -51,9 +51,16 @@ sub run {
         }
     }
 
-    if (check_screen('installed-product-incompatible', 10)) {
-        send_key 'alt-o';    # C&ontinue
-        record_soft_failure 'installed product incompatible';
+    if (check_screen(["installed-product-incompatible", "inst-betawarning"], 10)) {
+        send_key 'alt-o';
+        if (match_has_tag 'installed-product-incompatible') {
+            record_soft_failure 'installed product incompatible';
+        }
+        else {
+            if (!get_var("BETA")) {
+                record_soft_failure "Beta warning in non beta product";
```

Comment:
IMHO this whole `if (!get_var("BETA"))` block is a perfect candidate to move in a separate function. This way we could also avoid the different warnings for the same issue (see l#41)

Reply:
Thats a good idea, but shouldn't this be made in a separate PR to keep this ticket related?

Reply:
You introduce duplicate code here *because* of this ticket. I'd refuse to merge duplicate code at all - but lucky you, I've no merge rights anyway ;)